spec/features/activity_pages/activity_pages_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+describe 'Activity pages:' do
+  subject { page }
+
+  it 'should require authenticated users' do
+    Configuration.create(name: 'admin:password', value: 'rspec_pass')
+    visit project_activities_path(create(:project))
+    expect(current_path).to eq(login_path)
+    expect(page).to have_content('Access denied.')
+  end
+
+  context 'as authenticated user' do
+    describe 'index page', js: true do
+      let(:trackable) { create(:issue) }
+      let(:user) { create(:user) }
+
+      let(:create_activities) do
+        50.times do
+          activity = Activity.create(
+            user: user,
+            trackable_type: trackable.class,
+            trackable_id: trackable.id,
+            action: 'update',
+            created_at: Time.current - ((1..5).to_a.sample.days)
+          )
+        end
+      end
+
+      before do
+        create_activities
+        login_to_project_as_user
+        visit project_activities_path(current_project)
+      end
+
+      it 'shows paginated records' do
+        expect(page).to have_selector('.activity', count: Kaminari.config.default_per_page)
+      end
+
+      it 'shows unique date headers' do
+        activities_groups = Activity.order(created_at: :desc).limit(Kaminari.config.default_per_page).group_by do
+          |activity| activity.created_at.strftime(Activity::ACTIVITIES_STRFTIME_FORMAT)
+        end
```

Comment:
```suggestion
        activities_groups = Activity.order(created_at: :desc).limit(Kaminari.config.default_per_page).group_by do |activity|
           activity.created_at.strftime(Activity::ACTIVITIES_STRFTIME_FORMAT)
        end
```

You can also split the `activities_group` query into multiple lines instead of one chain for better readability.